Dev mvvm clazz edit

ParimSsme opened the pull request.

ParimSsme committed:
create clazzEnrolmentEdit screen with jetpack compose and react

ParimSsme committed:
create ScopedGrantEditScreenPreview screen with jetpack compose and react

ParimSsme committed:
create ClazzEdit screen with react

ParimSsme committed:
Merge branch 'dev-mvvm-primary' of https://github.com/UstadMobile/UstadMobile into dev-mvvm-clazzEdit

ParimSsme committed:
remove unrelated changes with dev-mvvm-primary except clazzEdit files

ParimSsme committed:
"Work in progress MVVM design."

ParimSsme committed:
"Work in progress MVVM design."

ParimSsme committed:
Merge branch 'dev-mvvm-primary' of https://github.com/UstadMobile/UstadMobile into dev-mvvm-clazzEdit

 Conflicts:
	app-android/src/main/java/com/ustadmobile/port/android/view/InviteViaLinkFragment.kt
	app-android/src/main/java/com/ustadmobile/port/android/view/RegisterMinorWaitForParentFragment.kt
	app-android/src/main/java/com/ustadmobile/port/android/view/SettingsFragment.kt
	app-android/src/main/java/com/ustadmobile/port/android/view/SiteTermsDetailFragment.kt
	app-react/src/main/kotlin/com/ustadmobile/hooks/useUstadScreens.kt
	app-react/src/main/kotlin/com/ustadmobile/view/InviteViaLinkScreen.kt
	app-react/src/main/kotlin/com/ustadmobile/view/RegisterAgeRedirectScreen.kt
	app-react/src/main/kotlin/com/ustadmobile/view/RegisterMinorWaitForParentScreen.kt
	app-react/src/main/kotlin/com/ustadmobile/view/SettingsScreen.kt
	app-react/src/main/kotlin/com/ustadmobile/view/SiteTermsDetailScreen.kt
	core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/InviteViaLinkViewModel.kt
	core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/RegisterMinorWaitForParentViewModel.kt
	core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/SiteTermsDetailViewModel.kt

ParimSsme committed:
merge branch dev-mvvm-primary into dev-mvvm-clazzEdit

ParimSsme committed:
"Work in progress MVVM design."

ParimSsme committed:
"fix(SchoolEditFragment.kt):use LazyColumn instead of scrollable column"

ParimSsme committed:
Merge branch 'dev-mvvm-primary' of https://github.com/UstadMobile/UstadMobile into dev-mvvm-clazzEdit

 Conflicts:
	app-react/src/main/kotlin/com/ustadmobile/hooks/useUstadScreens.kt

ParimSsme committed:
"refactor: replace UstadSwitchField with related codes"

ParimSsme committed:
Merge branch 'dev-mvvm-primary' of https://github.com/UstadMobile/UstadMobile into dev-mvvm-clazzEdit

ParimSsme committed:
merge branch dev-mvvm-primary into dev-mvvm-clazzEdit

ParimSsme committed:
"fix(ClazzEditFragment): fix displaying data in clazzSchedules list"

ParimSsme committed:
add UstadDragDropSwipeList in jetpack compose

ParimSsme committed:
Merge branch 'dev-mvvm-UstadDragDropSwipeList' of https://github.com/UstadMobile/UstadMobile into dev-mvvm-clazzEdit

ParimSsme committed:
"add UstadLazyColumnDragAndDrop in jetpack compose"

ParimSsme committed:
"add UstadLazyColumnDragAndDrop in jetpack compose"

ParimSsme committed:
"add UstadLazyColumnDragAndDrop in jetpack compose"

ParimSsme committed:
"add UstadLazyColumnDragAndDrop in jetpack compose"

ParimSsme committed:
Merge branch 'dev-mvvm-primary' of https://github.com/UstadMobile/UstadMobile into dev-mvvm-clazzEdit

 Conflicts:
	app-react/src/main/kotlin/com/ustadmobile/hooks/useUstadScreens.kt

ParimSsme committed:
"add UstadLazyColumnDragAndDrop in jetpack compose"

ParimSsme committed:
"fix listitem in schedule list"

ParimSsme committed:
"feature(ClazzEditFragment, ClazzEditScreen) add course block item"

ParimSsme committed:
Merge branch 'dev-mvvm-primary' of https://github.com/UstadMobile/UstadMobile into dev-mvvm-clazzEdit

ParimSsme committed:
"fix(ClazzEditScreen) set correct data to schedule list item"

ParimSsme committed:
Merge branch 'dev-mvvm-primary' of https://github.com/UstadMobile/UstadMobile into dev-mvvm-clazzEdit

ParimSsme committed:
"completed ClazzEdit screen with jetpack compose and react"

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt:
```diff
@@ -284,4 +317,376 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
+            items = uiState.courseBlockList
+        ) { courseBlock ->
+
+            val CONTENT_ENTRY_TYPE_ICON_MAP = mapOf(
```
This map is a constant - so let's move it outside of the composable function (for performance). We don't need to make a new copy of this map every time the function runs, and then repeat that for each block.

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt:
```diff
@@ -284,4 +317,376 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
```
Do we need the Spacer items when we have the padding? Doesn't padding already solve the problem for us?

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt:
```diff
@@ -284,4 +317,376 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
+
+            val courseBlockEditAlpha: Float = if (courseBlock.cbHidden) 0.5f else 1f
+
+            ListItem(
```
To implement the indent: the start padding of the ListItem should be 8dp * the indent level of the block.

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt:
```diff
@@ -284,4 +317,376 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
+            )
+        }
+
+        items(
```
better to use:
```
items(
 items = uiState.clazzSchedules
) { schedule ->
   
}
```

We don't need to access items by using an array index. I think this is clearer than using it as an index.

Also: the Schedules are part of a List, so they should use ListItem, not UstadDetailField.

ParimSsme committed:
"fix(ClazzEditFragment, ClazzEditScreen): fix course block item alpha"

ParimSsme committed:
"make changes from pull request comments"

ParimSsme committed:
"fix(ClazzEditFragment.kt, ClazzEditScreen.kt): fix indent level in courseBlock item"

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt:
```diff
@@ -284,4 +317,393 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
 
+}
+
+private val CONTENT_ENTRY_TYPE_ICON_MAP = mapOf(
```
This also needs to include the icon maps for CourseBlock as per CourseBlock.kt:

        const val BLOCK_MODULE_TYPE = 100

        const val BLOCK_TEXT_TYPE = 102

        const val BLOCK_ASSIGNMENT_TYPE = 103

        const val BLOCK_CONTENT_TYPE = 104

        const val BLOCK_DISCUSSION_TYPE = 105

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt:
```diff
@@ -284,4 +317,393 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
+        item {
+            ListItem(
+                modifier = Modifier.clickable {
+                    onClickAddCourseBlock()
```
Please add the plus icon for the add course block item.

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt:
```diff
@@ -284,4 +317,393 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
+                    " ${messageIdResource(schedule.scheduleDay)} " +
+                    " $fromTimeFormatted - $toTimeFormatted "
+
+            ListItem(
```
Please add an invisible icon (e.g. alpha = 0)  to the schedule ListItem so that the alignment matches with the add button.

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt:
```diff
@@ -284,4 +317,393 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
+
+            ListItem(
+                modifier = Modifier
+                    .padding(start = startPadding)
```
There needs to be an onClickCourseBlock event handler.

mikedawson committed:
ClazzEdit: Jetpack compose - standardize spacing. Correct icon handling.

mikedawson committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-clazzEdit

mikedawson committed:
ClazzEditScreen - React
Fix ListItems to be together in one list (including items themselves and add buttons)
Fix Schedule text display
Fix incorrect use of useFormattedTime within useFormattedDate (nested useMemo not allowed)

mikedawson committed:
Add composereorderable dependency.

ParimSsme committed:
Merge branch 'dev-mvvm-primary' of https://github.com/UstadMobile/UstadMobile into dev-mvvm-clazzEdit

ParimSsme committed:
Merge branch 'dev-mvvm-clazzEdit' of https://github.com/UstadMobile/UstadMobile into dev-mvvm-clazzEdit

mikedawson committed:
ClazzEdit - work in progress - implementation of Android drag/drop reordering.

mikedawson committed:
Merge branch 'dev-mvvm-clazzEdit' of github.com:UstadMobile/UstadMobile into dev-mvvm-clazzEdit

mikedawson committed:
Merge branch 'dev-mvvm-primary' into dev-mvvm-clazzEdit

mikedawson committed:
Update drag/drop items on ClazzEdit to prevent user from dragging/dropping items out of bounds.

mikedawson committed:
Add working version of React Easy Sort wrapper. Move JS wrappers into a dedicated package.

mikedawson committed:
Finish wrapper for react easy sort.

mikedawson committed:
Create CourseBlockUiState to manage visibility of items on popupmenu for ClazzEdit
